utils/peerutils.go

```diff
@@ -3,27 +3,38 @@ package utils
 import (
 	"errors"
 	"net"
+	"strings"
 
 	config "github.com/spf13/viper"
 )
 
 // FormRemotePeerAddress will check and validate peeraddress provided. It will
 // return an address of the form <ip:port>
 func FormRemotePeerAddress(peeraddress string) (string, error) {
-	var remotePeerAddress string
 
 	host, port, err := net.SplitHostPort(peeraddress)
-	if err != nil || host == "" {
-		return "", errors.New("Invalid peer address")
+	if err != nil {
+		// net.SplitHostPort() returns an error if port is missing.
+		if strings.HasPrefix(err.Error(), "missing port in address") {
```

Comment:
Did you try to find an alternate way to check for error here? There should be a better way to check for this error other than comparing strings.